src/components/app/MenuButtons/MetaInfo.js

```diff
@@ -0,0 +1,167 @@
+                      <span className="metaInfoLabel">Profile Version:</span>
+                      {meta.preprocessedProfileVersion}
+                    </div>
+                  ) : null}
+                </div>
+                <h2 className="arrowPanelSubTitle">Application</h2>
+                <div className="arrowPanelSection">
+                  {meta.product ? (
+                    <div className="metaInfoRow">
+                      <span className="metaInfoLabel">Name:</span>
+                      {meta.product}
+                    </div>
+                  ) : null}
+                  {meta.misc ? (
+                    <div className="metaInfoRow">
+                      <span className="metaInfoLabel">Version:</span>
+                      {_formatVersionNumber(meta.misc)}
+                    </div>
+                  ) : null}
+                  {meta.appBuildID ? (
+                    <div className="metaInfoRow">
+                      <span className="metaInfoLabel">Build ID:</span>
+                      {meta.sourceURL ? (
+                        <a
+                          href={meta.sourceURL}
+                          title={meta.sourceURL}
+                          target="_blank"
+                        >
+                          {meta.appBuildID}
+                        </a>
+                      ) : (
+                        meta.appBuildID
+                      )}
+                    </div>
+                  ) : null}
+                  {meta.extensions ? (
+                    <div className="metaInfoRow">
+                      <span className="metaInfoLabel">Extensions:</span>
+                      <ul className="metaInfoList">
+                        {_mapMetaInfoExtensionNames(meta.extensions.name)}
+                      </ul>
+                    </div>
+                  ) : null}
+                </div>
+                <h2 className="arrowPanelSubTitle">Platform</h2>
+                <div className="arrowPanelSection">
+                  {meta.platform ? (
+                    <div className="metaInfoRow">
+                      <span className="metaInfoLabel">Platform:</span>
+                      {meta.platform}
+                    </div>
+                  ) : null}
+                  {meta.oscpu ? (
+                    <div className="metaInfoRow">
+                      <span className="metaInfoLabel">OS:</span>
+                      {meta.oscpu}
+                    </div>
+                  ) : null}
+                  {meta.abi ? (
+                    <div className="metaInfoRow">
+                      <span className="metaInfoLabel">ABI:</span>
+                      {meta.abi}
+                    </div>
+                  ) : null}
+                </div>
+              </ArrowPanel>
+            }
+          />
+        </div>
+      );
+    }
+    return null;
+  }
+}
+
+function _mapMetaInfoExtensionNames(data: string[]): React.DOM {
```

Comment:
optional nit: it's not really necessary to prefix the functions with a `_`: they're not exported _so_ they're internal.